src/utils/muiMixout.js

```diff
@@ -0,0 +1,12 @@
+import {PropTypes} from 'react';
+import mixout, {remix, combine} from 'react-mixout';
+import pure from 'react-mixout-pure';
+import forwardContext from 'react-mixout-forward-context';
+
+export {mixout, remix, combine, pure, forwardContext};
+
+export const forwardTheme = forwardContext('muiTheme', {
+  validator: PropTypes.object.isRequired,
+});
+
+export const muiMixout = combine(pure, forwardTheme);
```

Comment:
We were debating around the `pure` logic with @nathanmarks earlier today.
1. We should probably not use a pure logic when a component takes children. Most of the time, it's going to be wasted CPU cycles. 
2. That mean that we should probably only use the pure logic for our simple leafs.
On the other hand, we are going to *memoize* the style with classe names, our render methods are going to be much faster. To a point where I'm wondering if having just one HOC for this logic worth it.
When we don't need one, e.g. the icons, I'm all in 👍.